src/Symfony/Component/Messenger/Bridge/Kafka/Transport/KafkaReceiver.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\Kafka\Transport;
+
+use Symfony\Component\Messenger\Bridge\Kafka\Stamp\KafkaReceivedMessageStamp;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+use Symfony\Component\Messenger\Transport\Serialization\PhpSerializer;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+class KafkaReceiver implements ReceiverInterface
+{
+    public function __construct(
+        private Connection $connection,
+        private SerializerInterface $serializer = new PhpSerializer(),
+    ) {
+    }
+
+    /** @psalm-return \Traversable<Envelope> */
+    public function get(): iterable
+    {
+        yield from $this->getEnvelope();
+    }
+
+    /** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
+    public function ack(Envelope $envelope): void
+    {
+        /** @var KafkaReceivedMessageStamp $transportStamp */
+        $transportStamp = $envelope->last(KafkaReceivedMessageStamp::class);
+
+        $this->connection->ack($transportStamp->message);
+    }
+
+    /** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
+    public function reject(Envelope $envelope): void
+    {
+        // no reject method for kafka transport
```

Comment:
From my experience, when the consumer is configured with

```yaml
enable.auto.offset.store: false
enable.auto.commit: false
```

the offset is only committed when explicitly doing `ack` call. This means in case of an error, the message is never acknowledged / the offset is never committed, hence the consumer will start lagging and won't be able to consume new messages. In some/many cases this may be intended, especially when the ordering of messages matter – but it's also possible that one does not want to block the consumer when there's a single failing message. Then if you have retry configured on the same transport, you start producing retry messages which never get consumed because the consumer is stuck on the initial message (given you only have 1 consumer or 1 consumer group).

I'd like to see a configuration variable to make ack call on reject, a bit similar to Redis transports' `delete_after_reject`, say `commit_offset_after_reject` or `ack_after_reject`.

---

Anyway, would be nice to see this Kafka transport in the core :+1: